rmgpy/reaction.py

```diff
@@ -1123,3 +990,62 @@ def copy(self):
         
         return other
 
+def _isomorphicSpeciesList(list1, list2, checkIdentical=False):
+    """
+    This method compares whether lists of species or molecules are isomorphic
+    or identical. It is used for the 'isIsomorphic' method of Reaction class.
+    It likely can be useful elswehere as well:
+        
+        list1 - list of species/molecule objects of reaction1
+        list2 - list of species/molecule objects of reaction2
+        checkIdentical - if true, uses the 'isIdentical' comparison
+                         if false, uses the 'isIsomorphic' comparison
+                         
+    Returns True if the lists are isomorphic/identical & false otherwise
+    """
+
+    def comparison_method(other1, other2, checkIdentical=checkIdentical):
+        if checkIdentical:
+            return other1.isIdentical(other2)
+        else:
+            return other1.isIsomorphic(other2)
+
+    if len(list1) == len(list2) == 1:
+        if comparison_method(list1[0], list2[0]):
+            return True
+    elif len(list1) == len(list2) == 2:
+        if comparison_method(list1[0], list2[0]) \
+                    and comparison_method(list1[1], list2[1]):
+            return True
+        elif comparison_method(list1[0], list2[1]) \
+                    and comparison_method(list1[1], list2[0]):
+            return True
+    elif len(list1) == len(list2) == 3:
```

Comment:
Or wait you don't even need to avoid copying the entries do you?
```
list1Temp = copy(list1)
list2Temp = copy(list2)
indexList = []
k = 0
while list1Temp != []:
   entry = list1Temp[k]
   for i,entry2 in enumerate(list2Temp):
       if comparison_method(entry,entry2):
           list1Temp.remove(entry)
           list2Temp.remove(entry2)
           break
   else: #loop doesn't break, something is missing
         return False
   k += 1
#while loop completed
return True
```